### DISCUSSION TITLE
Where is the timeout set for concurrent.futures.TimeoutError

### DISCUSSION THREAD

**Discussion by jordanjj8** [FreeOpcUa/python-opcua#1391] [Q&A] [answered]
Hey there, 

I have a function that tries to get a node but raises a timeout error when concurrent.futures.TimeoutError is raised. I found that this timeout is consistently 5 seconds and was wondering where and how can this value be changed?

for reference 

```
        self._client = Client(server_addr)
        self.connect()
        try:
            return self._client.get_node(node_id).get_value()
        except concurrent.futures.TimeoutError:
            raise TimeoutError
```

**Comment by AndreasHeine** [comment, marked as answer]
i assume this one: 
https://github.com/FreeOpcUa/python-opcua/blob/3012acf9047aad3acc2674e9008f0d1eee680c25/opcua/client/client.py#L95

**Reply by jordanjj8** [reply]
thank you very much

**Comment by oroulet** [comment]
but if your function takes more than 5s to answer you probably have an issue somewhere

**Reply by jordanjj8** [reply]
@oroulet could this be CPU depedent at all?

**Reply by oroulet** [reply]
CPU on plc yes
